data/decode.go

```diff
@@ -13,6 +13,100 @@ type NodeEdgeChildren struct {
 	Children []NodeEdgeChildren
 }
 
+type GroupedPoints struct {
+	Keyed    bool
+	IndexMax int
+	Points   []Point
+}
+
+var reflectValueT = reflect.TypeOf(reflect.Value{})
+
+func setValueFromPointGroup(g GroupedPoints, v reflect.Value) error {
```

Comment:
Maybe make this a method of the GroupedPoints struct?

Reply:
yes, I would probably lean that way.